mantidproject/mantid
# Speedup AlgorithmDialog system tests

**Description of work**
This PR speeds up the AlgorithmDialog Startup system tests by ensuring the test will directly call the `validateInputs` method (the thing we want it to be testing). Previously, the test was opening a Qt widget Algorithm Dialog in order to trigger the `validateInputs` method. This was very slow, and it was also not clear what the system tests were meant to be testing.

This PR also moves the tests into the `framework` system test folder as they no longer use any Qt.

The new implementation of these tests has also exposed some existing errors for some Mantid algorithms. These algorithms have been excluded from the tests temporarily and will be fixed in a subsequent PR. See this issue: https://github.com/mantidproject/mantid/issues/35686


**To test:**
To run the tests on this branch you can do:
```sh
./systemtest.bat -C DebugWithRelRuntime -R ValidateInputsTest
```
In total, this should take 10-20 seconds.

On the `main` branch, you can run the old tests:
```sh
./systemtest.bat -C DebugWithRelRuntime -R AlgorithmDialogs
```
This might take up to 10 minutes.


*There is no associated issue.*


#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
